dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java

```diff
@@ -914,4 +916,85 @@ private static boolean isVarArgsInvocation(
     return TypesUtils.getArrayDepth(ElementUtils.getType(lastParamElt))
         != TypesUtils.getArrayDepth(lastArgType);
   }
+
+  /**
+   * Find the declaration of the receiver of a method call in a method tree.
+   *
+   * <p>The receiver should appear in one of two places, either as a formal parameter to the method,
+   * or as a local variable.
+   *
+   * @param tree the method tree
+   * @param receiver the receiver for which to look up a declaration
+   * @return the declaration of the receiver of the method call, if found. Otherwise, null
+   */
+  public static @Nullable VariableTree getReceiverDeclarationInMethod(
+      @Nullable MethodTree tree, JavaExpression receiver) {
+    if (tree == null) {
+      return null;
+    }
+    List<? extends VariableTree> params = tree.getParameters();
+    for (VariableTree param : params) {
+      if (param.getName().toString().equals(receiver.toString())) {
+        return param;
+      }
+    }
+    for (StatementTree statement : tree.getBody().getStatements()) {
+      if (statement instanceof VariableTree) {
+        VariableTree localVariableTree = (VariableTree) statement;
+        if (localVariableTree.getName().toString().equals(receiver.toString())) {
+          return localVariableTree;
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Find the declaration of the receiver of a method call in a method tree.
```

Comment:
What does "in a method tree" refer to:  where the method call appears or where the declaration appears?  I think where the method call appears, but please clarify.